worker/wallet.js

```diff
@@ -284,3 +286,71 @@ export async function checkPendingWithdrawals (args) {
     }
   }
 }
+
+export async function checkWallet ({ data: { userId }, models }) {
+  const wallets = await models.wallet.findMany({
+    where: {
+      userId,
+      enabled: true
+    },
+    include: {
+      vaultEntries: true
+    }
+  })
+
+  const { hasRecvWallet: oldHasRecvWallet, hasSendWallet: oldHasSendWallet } = await models.user.findUnique({ where: { id: userId } })
+
+  const newHasRecvWallet = wallets.some(({ type, wallet }) => canReceive({ def: getWalletByType(type), config: wallet }))
+  const newHasSendWallet = wallets.some(({ vaultEntries }) => vaultEntries.length > 0)
+
+  await models.user.update({
+    where: { id: userId },
+    data: {
+      hasRecvWallet: newHasRecvWallet,
+      hasSendWallet: newHasSendWallet
+    }
+  })
```

Comment:
This should happen atomically with streak creation/update.

Reply:
Oh, right, will wrap it in a transaction

Reply:
Fixed in f5da4d52